plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -328,7 +328,8 @@ func (c *CloudWatch) fetchNamespaceMetrics() ([]types.Metric, error) {
 		for {
 			resp, err := c.client.ListMetrics(context.Background(), params)
 			if err != nil {
-				return nil, fmt.Errorf("failed to list metrics with params per namespace: %v", err)
+				fmt.Printf("failed to list metrics with namespace %s: %v", namespace, err)
+				break
```

Comment:
I don''t think this is the right fix for the following reasons:
1. You change the functionality as now the already added metrics are returned, while previously none was returned.
2. You should _never_ use `fmt.Printf` as it breaks logging.

Instead you probably want
```suggestion
				return nil, fmt.Errorf("failed to list metrics with namespace %s: %v", namespace, err)
```

Reply:
Two problems:
1. Error message not include namespace with problem. I set 11 namespaces and got error with 2 namespaces. So its take too time debug this issue.
2. If one of namespace get error from aws API why stop take other metrics, it's can be warning in logs and continue working

Reply:
Regarding 1.:
Agreed, that's why I proposed to simply add the namespace to the existing error message (see my suggestion).

Regarding 2.:
Your comment help me to understand your change better, therefore it would be nice if you can add it to the code and also to the PR description. I now see your point and it makes sense. So my proposal is to replace your `fmt.Printf` with `c.Log.Errorf` to get proper logging and add the comment on why we still continue on error...
Furthermore, do you see a way to retry to gather those metrics if the error is temporary (e.g. temporary loss of connection)?

**Please also add a meaningful PR description to the top!**

Reply:
Thanks for you comment
Done can you check now?